src/cata_imgui.cpp

```diff
@@ -667,3 +681,19 @@ void cataimgui::window::clear_filter()
         }
     }
```

Comment:
Which previously–pushed color would need to be popped?

Reply:
On my phone, but in line 424 you call PushOrPopColor, how do you know for sure when you're at the end of draw_colored_text that the last pushed color was popped?

Reply:
The colors it is pushing or popping come from the strings we are printing, which variously come from the JSON files the game loaded or are generated by other subsystems of the game. If they have unclosed color tags then one or more colors could indeed be left on the stack. But this will simply cause a bug report stating that some text is the wrong color. No other harm will be done. I think the stack is cleared after every frame, so it’s pretty harmless. 

I suppose if there were too many closing color tags than too many colors would be popped and eventually the stack would be emptied. That would cause an assertion from ImGui. Now that I think about it, however, I realize that we haven't hooked ImGui’s assertions up to the game’s nifty error reporting functionality, so those assertions are only effective in debug builds. We should fix that, but definitely in a different pull request.

Reply:
I just don't believe we can code for the best case scenario. If a text like "<color_yellow>test" is given as a parameter, we would push yellow onto the stack but never pop it back off. Errors like this must be handled. Maybe return a Boolean from your push or pop function, true if there is a color pushed onto the stack. If there is then it must be popped off by the end of the function, otherwise the function is not cleaning up after itself.

Reply:
I don’t think it’s a huge problem, but… I guess maybe it won’t hurt to check. A mere boolean wouldn’t do it though.